5.2.piece.of.cake.py

```diff
@@ -0,0 +1,21 @@
+def get_recipe_price(prices, optionals = None, **ingredients):
+    if optionals is None:
+        optionals = []
+    check = False
+    price = 0
+    for itemX in prices:
+        for itemY in optionals:
+            if itemX is itemY:
```

Comment:
there is no need to run over optionals in a loop.
read about the "in" operator

https://realpython.com/python-in-operator/